### DISCUSSION TITLE
[SOLVED] Problem with `use.lazyStateGetterSetter`

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#146] [General]
I removed all many previous messages. I have been close to giving up but I solved it. Solution was just to ignore rebuilds on build time.
```dart
if (!component._building) {
      component
        .._debug('forceUpdate()')
        ..forceUpdate();
    }
```

**Comment by GregoryConrad** [comment]
In the future, please try not to remove all of your comments; if you find they have bad/outdated info, it is normally best to just edit them and ~strike them through like this~ with `~my striked comment~` and add a note at the bottom that says it is out of date. It can make it hard to understand what is going on :)

That aside, is this for the react port? I realized I completely forgot to add a unit test for use.data in flutter_rearch so I added a test case and it worked with no issues (to my surprise): https://github.com/GregoryConrad/rearch-dart/commit/09bdb4425eac81d77d7bb6eb3e9ad19eae554dfe

I figured the markNeedsBuild would have thrown an assertion error during the build, but it apparently did not. Maybe since the widget is already building (or, in other words, is a descendant of itself, based on the markNeedsBuild source code), it is allowed.

Since setState/rebuild/render/forceUpdate/markNeedsBuild/whateverItsCalledOverInReact might not behave the same way as markNeedsBuild in Flutter, chances are you may need a `_isCurrentlyBuilding` flag that will absorb any calls to the render function and just call the passed in `sideEffectMutation` (if not null).

> Solution was just to ignore rebuilds on build time

Yup, if the markNeedsBuild behavior is different in Flutter than the equivalent in React, this is what you would need (and also what I figured I'd need over in `flutter_rearch`, but for some reason, it is working as-is without that)

**Comment by busslina** [comment]
> In the future, please try not to remove all of your comments.

Sure.

> That aside, is this for the react port?

Yes.

>  figured the markNeedsBuild would have thrown an assertion error during the build.

Also it's weird because it's marked as `dirty` but it doesn't get rebuilt. I tried to understand the Flutter implementation but is a chaos for me:
```
/// Returns true if the element has been marked as needing rebuilding.
///
/// The flag is true when the element is first created and after
/// [markNeedsBuild] has been called. The flag is reset to false in the
/// [performRebuild] implementation.
bool get dirty => _dirty;
bool _dirty = true;
```

> the markNeedsBuild behavior is different in Flutter than the equivalent in React

I like how finally it works because it works simple and sync. I'm migrating one project to test better and then I will try to clean my PR. E.G. I discussed with you about changing capsule state from a capsule listener (Websocket state) and you adviced me that I needed to do it asynchornously, but with the addition of `use.data` it is supposed to work synchronously. I will test it too
